app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -2896,6 +2903,10 @@ class BrowserTabFragment :
         }
     }
 
+    private fun showSitePermissionsDialog(permissionsToRequest: Array<String>, request: PermissionRequest) {
+        sitePermissionsDialogLauncher.askForSitePermission(requireContext(), webView?.url ?: "", permissionsToRequest, request)
```

Comment:
Using `requireContext` will crash the app if no context exists, you could do `context?.let {}` instead.

nit: you can also do `webView?.url.orEmpty()`

Reply:
Thanks for the suggestions! I will change it as it is much safer